File: solutions/Leetcode_917/leetcode_917.py
class Solution1:
    def reverse_only_letters(self, s: str) -> str:
        left = 0
        right = len(s) - 1
        s = list(s)
        while left < right:
            if not s[left].isalpha():
                left += 1
            elif not s[right].isalpha():
                right -= 1
            else:
                s[left], s[right] = s[right], s[left]
                left += 1
                right -= 1
        return ''.join(s)

File: solutions/Leetcode_917/leetcode_917.cpp
#include <string>

class Solution1
{
public:
    std::string reverseOnlyLetters(std::string s)
    {
        size_t left = 0;
        size_t right = s.size() - 1;
        while (left < right)
        {
            if (!std::isalpha(s[left]))
                left++;
            else if (!std::isalpha(s[right]))
                right--;
            else
            {
                std::swap(s[left], s[right]);
                left++;
                right--;
            }
        }
        return s;
    }
};
